test/commands/force/data/bulk/status.test.ts

```diff
@@ -4,136 +4,196 @@
+    },
+    prependListener(eventName: string | symbol, listener: (...args: any[]) => void): Job<any, BulkOperation> {
+      throw new Error('Function not implemented.');
+    },
+    prependOnceListener(eventName: string | symbol, listener: (...args: any[]) => void): Job<any, BulkOperation> {
+      throw new Error('Function not implemented.');
+    },
+    eventNames(): Array<string | symbol> {
+      throw new Error('Function not implemented.');
+    },
+  };
+  beforeEach(async () => {
+    await $$.stubAuths(testOrg);
+    await config.load();
+    stdoutSpy = $$.SANDBOX.stub(process.stdout, 'write');
+
+    $$.SANDBOXES.CONNECTION.stub(Bulk.prototype, 'job').returns({
+      id: '75054000006yv68AAA',
+      state: 'Open',
+      options: {},
+      type: null,
+      operation: null,
+      list: async () =>
+        Promise.resolve([
           {
-            apexProcessingTime: '0',
-            apiActiveProcessingTime: '55',
-            createdDate: '2020-11-30T16:14:21.000Z',
             id: '751540000070aNsAAI',
             jobId: '75054000006yv68AAA',
-            numberRecordsFailed: '1',
-            numberRecordsProcessed: '1',
             state: 'Completed',
+            createdDate: '2020-11-30T16:14:21.000Z',
             systemModstamp: '2020-11-30T16:14:21.000Z',
+            numberRecordsProcessed: '1',
+            numberRecordsFailed: '1',
             totalProcessingTime: '185',
+            apiActiveProcessingTime: '55',
+            apexProcessingTime: '0',
+            // won't appear on the real API response, but jsforce says it's required
+            stateMessage: '',
           },
-        ],
-        status: 0,
-      });
+        ]),
+      ...unusedInterfaceParts,
     });
+  });
+
+  afterEach(async () => {
+    $$.restore();
+  });
+
+  it('will fail with the correct error message for not found', async () => {
+    const cmd = new Status(
+      [
+        '--targetusername',
+        'test@org.com',
+        '--batchid',
+        '751540000070Q4RAAU',
+        '--jobid',
+        '75054000006ybyHAAQ',
+        '--json',
+      ],
+      config
+    );
+
+    // eslint-disable-next-line no-underscore-dangle
+    await cmd._run();
+    const result = JSON.parse(stripAnsi(stdoutSpy.args.flat().join(''))) as {
+      message: string;
+      status: number;
+      exitCode: number;
+    };
+    expect('exitCode' in result).to.equal(true);
+    expect(result.message).to.equal('Unable to find batch 751540000070Q4RAAU for job 75054000006ybyHAAQ.');
+    expect(result.exitCode).to.equal(1);
+  });
+
+  it('will successfully find the batchid and jobid', async () => {
+    const cmd = new Status(
+      [
+        '--targetusername',
+        'test@org.com',
+        '--batchid',
+        '751540000070aNsAAI',
+        '--jobid',
+        '75054000006yv68AAA',
+        '--json',
+      ],
+      config
+    );
+
+    // eslint-disable-next-line no-underscore-dangle
+    const result = await cmd._run();
```

Comment:
You might be able to avoid disabling eslint if you do this:

`await Status.run(args, config)`

Reply:
that does avoid the eslint, but oclif has the result typed as `any`
<img width="826" alt="Screen Shot 2022-11-14 at 9 36 09 AM" src="https://user-images.githubusercontent.com/4261788/201701134-7418d689-2994-4268-8819-1f0fd935efcb.png">

I'm gonna convert the ones that don't use the result.